test/checkpoint/checkpointer.c

```diff
@@ -373,6 +373,122 @@ real_checkpointer(THREAD_DATA *td)
     return (0);
 }
 
+/*
+ * prepare_discover --
+ *     Claim pending prepared transactions, do a checkpoint then verify the consistency of the data
+ *     post claiming.
+ */
+void
+prepare_discover(WT_CONNECTION *conn, THREAD_DATA *td)
+{
+    WT_UNUSED(td);
+    /*
+     * Since RTS is not ran with precise checkpoint, we need to use prepare discover cursor to claim
+     * all pending prepared transactions. When precise checkpoint is not configure, there's no need
+     * to run prepare discover.
+     */
+    if (!g.precise_checkpoint)
```

Comment:
Should we check preserve_prepared instead of precise_checkpoint?

Reply:
I understand. But what about the case that we only enable precise_checkpoint but not preserve_prepared? I think may be we should separate the two.

Reply:
ok, sure. but I think in test format we have two different configs?